R/Settings.R

```diff
@@ -89,300 +249,150 @@ addModuleSpecifications <- function(analysisSpecifications, moduleSpecifications
-#' @param integer64AsNumeric         Logical: should 64-bit integers be converted to numeric (double) values? If FALSE 64-bit integers will be represented using `bit64::integer64`.  Default is TRUE
 #'
 #' @return
 #' An object of type `ExecutionSettings`.
 #'
 #' @export
-createResultsExecutionSettings <- function(resultsConnectionDetailsReference,
-                                           resultsDatabaseSchema,
+createResultsExecutionSettings <- function(resultsDatabaseSchema,
                                            workFolder,
                                            resultsFolder,
                                            logFileName = file.path(resultsFolder, "strategus-log.txt"),
-                                           minCellCount = 5,
-                                           integerAsNumeric = getOption("databaseConnectorIntegerAsNumeric", default = TRUE),
-                                           integer64AsNumeric = getOption("databaseConnectorInteger64AsNumeric", default = TRUE)) {
+                                           minCellCount = 5) {
   errorMessages <- checkmate::makeAssertCollection()
-  checkmate::assertCharacter(resultsConnectionDetailsReference, len = 1, add = errorMessages)
   checkmate::assertCharacter(resultsDatabaseSchema, len = 1, add = errorMessages)
   checkmate::assertCharacter(workFolder, len = 1, add = errorMessages)
   checkmate::assertCharacter(resultsFolder, len = 1, add = errorMessages)
   checkmate::assertCharacter(logFileName, len = 1, add = errorMessages)
   checkmate::assertInt(minCellCount, add = errorMessages)
-  checkmate::assertLogical(integerAsNumeric, max.len = 1, add = errorMessages)
-  checkmate::assertLogical(integer64AsNumeric, max.len = 1, add = errorMessages)
   checkmate::reportAssertions(collection = errorMessages)
 
   # Normalize paths to convert relative paths to absolute paths
   workFolder <- normalizePath(workFolder, mustWork = F)
   resultsFolder <- normalizePath(resultsFolder, mustWork = F)
   logFileName <- normalizePath(logFileName, mustWork = F)
 
-  executionSettings <- list(
-    resultsConnectionDetailsReference = resultsConnectionDetailsReference,
-    resultsDatabaseSchema = resultsDatabaseSchema,
-    workFolder = workFolder,
-    resultsFolder = resultsFolder,
-    logFileName = logFileName,
-    minCellCount = minCellCount,
-    integerAsNumeric = integerAsNumeric,
-    integer64AsNumeric = integer64AsNumeric
-  )
+  executionSettings <- list()
+  for (name in names(formals(createResultsExecutionSettings))) {
+    executionSettings[[name]] <- get(name)
+  }
   class(executionSettings) <- c("ResultsExecutionSettings", "ExecutionSettings")
   return(executionSettings)
 }
 
-
-
-# Note: assuming connectionDetails objects remain stable across the various module
-# versions.
-
-#' Store connection details in a secure location
+#' Create Results Data Model Settings
```

Comment:
A couple of thoughts on the data model settings & functions:

- An implicit assumption with the work in this branch is that there is an `renv.lock` file that controls which version of the HADES package is in use when running Strategus. As a result, some of the modules are using a package's `createResultsDataModel` function which then creates the data model from the internally stored results data model specification vs. the one that is written to the `resultsFolder`. I think this is fine but could be confusing to supply a `resultsFolder` as a setting when it is not necessary for creating the results data model.
- `createResultsUploadSettings` currently has a parameter `purgeSiteDataBeforeUploading = FALSE` which corresponds to the [RMM uploadResults](https://ohdsi.github.io/ResultModelManager/reference/uploadResults.html) function. I'm going to remove this to simplify the settings & functions as will be described in the points below.
- I will collapse the `createResultsUploadSettings` and `createResultsDataModelSettings` into a single function`createResultsDataModelSettings` since these will be identical.
- `createResultsDataModel` and `uploadResults` should be simplistic - we should not attempt to purge site data in the module. Instead, more advanced results data management should be out of the scope of Strategus' module approach. So if you've already created a results data model and re-attempt this operation, it should fail. Same goes for uploading results: if you re-attempt to upload the same data it fails.

Reply:
Changes made in https://github.com/OHDSI/Strategus/pull/145/commits/1a68ae57e86214eced54004898d1b134e38a11b6